File: armstrong.cpp
#include<iostream>
#include<math.h>
using namespace std;

int main()
{
  int n;
  cin>>n;
  int sum=0,rem,q=n,h;
  int digits=0;
  while(q>0)
  {
    q=q/10;
    digits++;
  }
  int k=n;
  while(k!=0)
  {
    rem=k%10;
    h=pow(rem,digits);
    sum=sum+h;
    k=k/10;
  }
  if(sum==n)
    cout<<"\nIS AN ARMSTRONG NUMBER";
  else
    cout<<"\nNOT AN ARMSTRONG NUMBER";
}

File: armstrong.py
num = int(input("Enter a number: "))
sum = num
a = len(str(num))
sum1 = 0

while(num !=0 ):
    b = num %10
    sum1 = sum1+(b**a)
    num = num // 10

if(sum == sum1):
    print("The number is armstrong")
else:
    print("The number is",sum, "not am armstrong number")
